src/FakeItEasy/Creation/DefaultFakeCreatorFacade.cs

```diff
@@ -35,6 +35,14 @@ public T CreateFake<T>(Action<IFakeOptions<T>> optionsBuilder)
             return (T)this.fakeAndDummyManager.CreateFake(typeof(T), options => optionsBuilder((IFakeOptions<T>)options));
         }
 
+        public object CreateFake(Type typeOfFake, Action<IFakeOptions> optionsBuilder)
+        {
+            Guard.AgainstNull(typeOfFake, nameof(typeOfFake));
+            Guard.AgainstNull(optionsBuilder, nameof(optionsBuilder));

```

Comment:
No problem, fully understandable. I believe we're using null guards in a lot of redundant places. It may be worth doing a cross cutting sweep of all of them at some point and we should consider our approach then.

For this PR, I think it's fine as-is.